src/javascript/app_2/App/Components/Form/TimePicker/time_picker.jsx

```diff
@@ -0,0 +1,123 @@
+import { observer }       from 'mobx-react';
+import PropTypes          from 'prop-types';
+import React              from 'react';
+import TimePickerDropdown from './time_picker_dropdown.jsx';
+import InputField         from '../input_field.jsx';
+
+class TimePicker extends React.PureComponent {
```

Comment:
@morteza-binary , shouldn't this be `React.Component` instead? Correct me if I'm wrong.

Reply:
@zakibinary Morteza was the one who suggest me to use PureComponent instead.

Reply:
@ahkuma , any updates on this? `PureComponent` will give an error/warning since it calls `shouldComponentUpdate()`.

Reply:
i think better to use React.Component if we're going to attach an observer to this component

Reply:
@zakibinary @kellybinary I am just going to use `Components` as it seems to be the best way. If we were to use `observer` with `PureComponent` it's going to apply a `shouldComponentUpdate()` anyway.